File: backend/src/main/java/org/pupille/backend/mysql/screening/TerminDTOWithFilmAndReiheDTOGallery.java
package org.pupille.backend.mysql.screening;

import org.pupille.backend.mysql.film.Film;
import org.pupille.backend.mysql.reihe.Reihe;
import org.pupille.backend.mysql.reihe.ReiheDTOGallery;
import org.pupille.backend.mysql.termin.Termin;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

public record TerminDTOWithFilmAndReiheDTOGallery(
        Long tnr,
        LocalDateTime vorstellungsbeginn,
        String titel,
        String kurztext,
        String besonderheit,
        String bild,
        String sonderfarbe,
        Short veroeffentlichen,
        List<FilmDTOGallery> mainfilms,
        Set<ReiheDTOGallery> reihen
) {
    public TerminDTOWithFilmAndReiheDTOGallery(Termin termin, List<Film> mainfilms, Set<Reihe> reihen) {
        this(
                termin.getTnr(),
                termin.getVorstellungsbeginn(),
                termin.getTitel(),
                termin.getKurztext(),
                termin.getBesonderheit(),
                termin.getBild(),
                termin.getSonderfarbe(),
                termin.getVeroeffentlichen(),
                mainfilms.stream()
                        .map(FilmDTOGallery::new)
                        .toList(),
                reihen.stream()
                        .map(ReiheDTOGallery::new)
                        .collect(Collectors.toSet())
        );
    }
}

File: frontend/src/types/TerminDTOWithFilmAndReiheDTOGallery.ts
import FilmDTOGallery from "./FilmDTOGallery.ts";
import ReiheDTOGallery from "./ReiheDTOGallery.ts";

// this one mirrors TerminDTOWithFilmAndReiheDTOGallery
interface TerminDTOWithFilmAndReiheDTOGallery {
    tnr: number;
    vorstellungsbeginn?: string;
    titel?: string | null;
    kurztext?: string | null;
    besonderheit?: string| null;
    bild?: string| null;
    sonderfarbe?: string | null;
    veroeffentlichen?: number | null;
    mainfilms: FilmDTOGallery[];
    reihen: ReiheDTOGallery[];
}

export default TerminDTOWithFilmAndReiheDTOGallery;